### PR TITLE
Add heatmap plot

### PR DESCRIPTION
**Pull request by natifridman**
Add heatmap abstract class for creating plots

### CODE REVIEW

File: src/vse_sync_pp/heatmap.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        plt.xlabel(self._xlabel)
+        plt.ylabel(self._ylabel)
+
+        # Loop over data dimensions and create text annotations.
+        for i in range(len(self._x_ticks)):
+            for j in range(len(self._y_ticks)):
+                ax.text(j, i, np_data[i, j],
+                        ha="center", va="center", color="white")
+
+        for row, col in self._unallowed_cells:
```
Comment (spenced) [inline src/vse_sync_pp/heatmap.py:47]: Recommend we document what the caller should supply in `unallowed_cells` and how its used.

File: src/vse_sync_pp/heatmap.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+
+class Heatmap():
+    """
+    Heatmap visualization for state transitions
+    Input parameters:
+    x_ticks - x axis ticks
+    y_ticks - y axis ticks
+    title - heatmap title
+    unallowed_cells - a list of unallowed cells to color in red.
```
Comment (jnunyez) [inline src/vse_sync_pp/heatmap.py:17]:
```suggestion
    unallowed_cells - a list of cells to color in red representing unallowed stated transitions.
```

File: src/vse_sync_pp/heatmap.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+"""Heatmap data parsed from log messages from a single source."""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.patches as patches
+
+
+class Heatmap():
+    """
+    Heatmap visualization for state transitions
```
Comment (jnunyez) [inline src/vse_sync_pp/heatmap.py:12]: An attempt at trying to generalize the description of the heatmap:
```suggestion
    Heatmap to visualize data values across two axis variables as a grid of colored squares.
```

**Review by jnunyez — COMMENTED** [review]
Added a few suggestions to make the class documentation more generic. The rest LGTM.

File: src/vse_sync_pp/heatmap.py (hunk 2 above)
Comment (natifridman) [reply src/vse_sync_pp/heatmap.py]: Done

File: src/vse_sync_pp/heatmap.py (hunk 3 above)
Comment (natifridman) [reply src/vse_sync_pp/heatmap.py]: Done

**Review by jnunyez — APPROVED** [review]

**Comment by jnunyez** [comment]
@spenced anything blocking us to get this merged?

**Pull request merged**